lib/beaker-pe/pe-client-tools/executable_helper.rb

```diff
@@ -0,0 +1,153 @@
+        def puppet_db_on(*args, &block)
+          Private.new.tool(:db, *args, &block)
+        end
+
+        # puppet-query helper win/lin/osx
+        # @param [BEAKER::Host] host The SUT that should run the puppet-query command
+        # @param [String] args The arguments to puppet-query
+        # @param [Hash] opts options hash to the Beaker Command
+        # @param [Block] &block optional block
+        def puppet_query_on(*args, &block)
+          Private.new.tool(:query, *args, &block)
+        end
+
+        # Logs a user in on a SUT with puppet-access/RBAC API (windows)
+        # @param [Beaker::Host] host The SUT to perform the login on
+        # @param [Scooter::HttpDispatchers::ConsoleDispatcher] credentialed_dispatcher A Scooter dispatcher that has credentials for the user
+        # @option attribute_hash [String] :name The environment variable
+        # @option attribute_hash [String] :default The default value for the environment variable
+        # @option attribute_hash [String] :message A message describing the use of this variable
+        # @option attribute_hash [Boolean] :required Used internally by CommandFlag, ignored for a standalone EnvVar
+        def login_with_puppet_access_on(host, credentialed_dispatcher, opts={})
+
+          lifetime = opts[:lifetime] || nil
+          unless host.platform =~ /win/
+
+            user = credentialed_dispatcher.credentials.login
+            password = credentialed_dispatcher.credentials.password
+            puppet_access_on(host, 'login', {:stdin => "#{user}\n#{password}\n"})
+          else
+
+            # this is a hack
+            # puppet-access needs to support alternative to interactive login
+            # create .puppetlabs dir
+            cmd = Beaker::Command.new('echo', ['%userprofile%'], :cmdexe => true)
+            user_home_dir = host.exec(cmd).stdout.chomp
+            win_token_path =  "#{user_home_dir}\\.puppetlabs\\"
+            host.exec(Beaker::Command.new('MD', [win_token_path.gsub('\\', '\\\\\\')], :cmdexe => true), :accept_all_exit_codes => true)
+
+            token = credentialed_dispatcher.acquire_token_with_credentials(lifetime)
+            create_remote_file(host, "#{win_token_path}\\token", token)
+          end
+        end
+
+        class Private
+
+          include Beaker::DSL
+          include Beaker::DSL::Wrappers
+          include Beaker::DSL::Helpers::HostHelpers
+          include Beaker::DSL::Patterns
+
+          attr_accessor :logger
+
+          def tool(tool, *args, &block)
+
+            host = args.shift
+            @logger = host.logger
+            options = {}
+            options.merge!(args.pop) if args.last.is_a?(Hash)
+
+            if host.platform =~ /win/i
+
+              program_files = host.exec(Beaker::Command.new('echo', ['%PROGRAMFILES%'], :cmdexe => true)).stdout.chomp
+              client_tools_dir = "#{program_files}\\#{['Puppet Labs', 'Client', 'tools', 'bin'].join('\\')}\\"
```

Comment:
doesn't `PEClientTools::ConfigFileHelper` figure this out, with some pain, above?